File: Swift-C++-Testsuite/test/arm/Swift/UnownedReferences.swift
@main
struct MyMain {

    static func main() {
        var john: Customer?
        john = Customer(name: "John Appleseed")
        john!.card = CreditCard(number: 1234_5678_9012_3456, customer: john!)
        john = nil
    }
}

class Customer {
    let name: String
    var card: CreditCard?
    init(name: String) {
        self.name = name
    }
    deinit { print("\(name) is being deinitialized") }
}

class CreditCard {
    let number: UInt64
    unowned let customer: Customer
    init(number: UInt64, customer: Customer) {
        self.number = number
        self.customer = customer
    }
    deinit { print("Card #\(number) is being deinitialized") }
}

File: AutoTranspiledTests/Swift-to-Cpp/try-1/UnownedReferences.cpp

 #include <iostream>

using namespace std;

class Customer {
public:
  string name;
  CreditCard* card;
  Customer(string name){
    this->name = name;
  }
  ~Customer(){
    cout << name << " is being deinitialized" << endl;
  }
};

class CreditCard {
public:
  long number;
  Customer customer;
  CreditCard(long number, Customer customer){
    this->number = number;
    this->customer = customer;
  }
  ~CreditCard() {
    cout << "Card #" << number << " is being deinitialized" << endl;
  }
};

int main(){
  Customer *john = new Customer("John Appleseed");
  john->card = new CreditCard(1234567890123456, *john);
  delete john;
  return 0;

}